# Title & Introduction

## 1. Import Packages and Clean Data

### 1a. Import Packages and Read in Data Frame

In [1]:
# Import Packages
import pandas as pd

In [9]:
# Read in dataset
url = "https://raw.githubusercontent.com/kellyshreeve/Integrated_Project_1/main/moved_games.csv"
df = pd.read_csv(url)

In [8]:
# View first 15 rows of the dataset 
df.head(15)

Name Platform  Year_of_Release         Genre  \
0                    Wii Sports      Wii           2006.0        Sports   
1             Super Mario Bros.      NES           1985.0      Platform   
2                Mario Kart Wii      Wii           2008.0        Racing   
3             Wii Sports Resort      Wii           2009.0        Sports   
4      Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing   
5                        Tetris       GB           1989.0        Puzzle   
6         New Super Mario Bros.       DS           2006.0      Platform   
7                      Wii Play      Wii           2006.0          Misc   
8     New Super Mario Bros. Wii      Wii           2009.0      Platform   
9                     Duck Hunt      NES           1984.0       Shooter   
10                   Nintendogs       DS           2005.0    Simulation   
11                Mario Kart DS       DS           2005.0        Racing   
12  Pokemon Gold/Pokemon Silver       GB           1999.0  Role-Playing   
13                      Wii Fit      Wii           2007.0        Sports   
14           Kinect Adventures!     X360           2010.0          Misc   

    NA_sales  EU_sales  JP_sales  Other_sales  Critic_Score User_Score Rating  
0      41.36     28.96      3.77         8.45          76.0          8      E  
1      29.08      3.58      6.81         0.77           NaN        NaN    NaN  
2      15.68     12.76      3.79         3.29          82.0        8.3      E  
3      15.61     10.93      3.28         2.95          80.0          8      E  
4      11.27      8.89     10.22         1.00           NaN        NaN    NaN  
5      23.20      2.26      4.22         0.58           NaN        NaN    NaN  
6      11.28      9.14      6.50         2.88          89.0        8.5      E  
7      13.96      9.18      2.93         2.84          58.0        6.6      E  
8      14.44      6.94      4.70         2.24          87.0        8.4      E  
9      26.93      0.63      0.28         0.47           NaN        NaN    NaN  
10      9.05     10.95      1.93         2.74           NaN        NaN    NaN  
11      9.71      7.47      4.13         1.90          91.0        8.6      E  
12      9.00      6.18      7.20         0.71           NaN        NaN    NaN  
13      8.92      8.03      3.60         2.15          80.0        7.7      E  
14     15.00      4.89      0.24         1.69          61.0        6.3      E

In [4]:
# Print dataset info
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16715 entries, 0 to 16714
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16713 non-null  object 
 1   Platform         16715 non-null  object 
 2   Year_of_Release  16446 non-null  float64
 3   Genre            16713 non-null  object 
 4   NA_sales         16715 non-null  float64
 5   EU_sales         16715 non-null  float64
 6   JP_sales         16715 non-null  float64
 7   Other_sales      16715 non-null  float64
 8   Critic_Score     8137 non-null   float64
 9   User_Score       10014 non-null  object 
 10  Rating           9949 non-null   object 
dtypes: float64(6), object(5)
memory usage: 1.4+ MB


This dataset has a total of 16715 entries. There are missing values in Name, Year_of_Release, Genre, Critic_Score, User_Score, and Rating. Year_of_Release needs to be changed to int data type and and User_Score needs to be changed to float data type.

In [ ]:
# Change variable names to snake case


### 1b. Change Data Types

In [5]:
# Print unique values of 'user_score'
df['User_Score'].unique()

array(['8', nan, '8.3', '8.5', '6.6', '8.4', '8.6', '7.7', '6.3', '7.4',
       '8.2', '9', '7.9', '8.1', '8.7', '7.1', '3.4', '5.3', '4.8', '3.2',
       '8.9', '6.4', '7.8', '7.5', '2.6', '7.2', '9.2', '7', '7.3', '4.3',
       '7.6', '5.7', '5', '9.1', '6.5', 'tbd', '8.8', '6.9', '9.4', '6.8',
       '6.1', '6.7', '5.4', '4', '4.9', '4.5', '9.3', '6.2', '4.2', '6',
       '3.7', '4.1', '5.8', '5.6', '5.5', '4.4', '4.6', '5.9', '3.9',
       '3.1', '2.9', '5.2', '3.3', '4.7', '5.1', '3.5', '2.5', '1.9', '3',
       '2.7', '2.2', '2', '9.5', '2.1', '3.6', '2.8', '1.8', '3.8', '0',
       '1.6', '9.6', '2.4', '1.7', '1.1', '0.3', '1.5', '0.7', '1.2',
       '2.3', '0.5', '1.3', '0.2', '0.6', '1.4', '0.9', '1', '9.7'],
      dtype=object)

In [6]:
df['Year_of_Release'].unique()

array([2006., 1985., 2008., 2009., 1996., 1989., 1984., 2005., 1999.,
       2007., 2010., 2013., 2004., 1990., 1988., 2002., 2001., 2011.,
       1998., 2015., 2012., 2014., 1992., 1997., 1993., 1994., 1982.,
       2016., 2003., 1986., 2000.,   nan, 1995., 1991., 1981., 1987.,
       1980., 1983.])

### 1c. Check for Missing Values

### 1d. Check for Duplicates

### 1e. Add Additional Features

## 2. Analyze the data

## 3. Create a User Profile for Each Region

## 4. Test Hypotheses

## 5. Conclusion